esphome/components/dxs238xw/switch/__init__.py

```diff
@@ -0,0 +1,76 @@
+from esphome.components import switch
+import esphome.config_validation as cv
+import esphome.codegen as cg
+
+from esphome.const import (
+    CONF_ID,
+    CONF_ICON,
+    CONF_ENTITY_CATEGORY,
+    ENTITY_CATEGORY_CONFIG,
+)
+
+from .. import dxs238xw_ns, CONF_DXS238XW_ID, SmIdEntity, DXS238XW_COMPONENT_SCHEMA
+
+DEPENDENCIES = ["uart"]
+
+dxs238xwSwitch = dxs238xw_ns.class_("Dxs238xwSwitch", switch.Switch)
+
+ENERGY_PURCHASE_STATE = "energy_purchase_state"
+METER_STATE = "meter_state"
+DELAY_STATE = "delay_state"
+
+TYPES = {
+    ENERGY_PURCHASE_STATE: (
+        switch.SWITCH_SCHEMA.extend(
+            {
+                cv.GenerateID(): cv.declare_id(dxs238xwSwitch),
+                cv.Optional(CONF_ICON, default="mdi:lightning-bolt"): cv.icon,
+                cv.Optional(
+                    CONF_ENTITY_CATEGORY, default=ENTITY_CATEGORY_CONFIG
+                ): cv.entity_category,
+            }
+        ),
```

Comment:
This should simplify the switch code a bit simiilar to how sensor is done.
Also all icon strings should be added in `esphome/const.py`

```suggestion
        switch.switch_schema(
            dxs238xwSwitch,
            icon=ICON_LIGHTNING_BOLT,
            entity_category=ENTITY_CATEGORY_CONFIG
        ),
```

Reply:
At first, I try to define everything in the same way as I did in sensor, but as of today it is not possible. For example, on the switch it wouldn't work since "switch_schema" doesn't define the "icon" option like sensor does.

![Captura de pantalla 2022-08-31 103831](https://user-images.githubusercontent.com/6945715/187706144-60bcea9f-f47b-454c-99a8-f1a38437111f.png)

In the case of button, is not defined "class_".
In the case of number, "number_schema" doesn't even exist.

I didn't want to modify other files outside of my component, so for the number, switch, and button I adapted to what exists today.

Reply:
I noticed that they recently changed the definition of switch_schema. So I update according to your indication.

The number and buttons are expected to update, so update the same way as the switch?

Reply:
Yeah sorry I did it recently =)
Dont worry about the ones that are not done yet like number/button. I will go through and update each platform when I make the core changes.

Reply:
ok